src/dreammaker/objtree.rs

```diff
@@ -890,12 +909,14 @@ impl ObjectTree {
             is_static,
             is_const,
             is_tmp,
+            is_final,
             type_path,
         };
         var_type.suffix(&suffix);
 
-        let symbols = &mut self.symbols;
+        let symbolid = self.symbols.allocate();
```

Comment:
Old code allocates a symbol ID only if `is_declaration` is true and there wasn't already an entry, now it's unconditional. Seems like an undesirable change to me.